app/src/main/java/org/greenstand/android/TreeTracker/fragments/LoginFragment.kt

```diff
@@ -0,0 +1,200 @@
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
+import org.greenstand.android.TreeTracker.R
+import org.greenstand.android.TreeTracker.activities.CameraActivity
+import org.greenstand.android.TreeTracker.application.Permissions
+import org.greenstand.android.TreeTracker.application.TreeTrackerApplication
+import org.greenstand.android.TreeTracker.database.entity.PlanterDetailsEntity
+import org.greenstand.android.TreeTracker.utilities.ImageUtils
+import org.greenstand.android.TreeTracker.utilities.ValueHelper
+import timber.log.Timber
+
+class LoginFragment : Fragment(){
+    var phoneNumberEntered: String? = null
+    var emailEntered: String? = null
+    private var mPhotoPath: String? = null
+    private var planterId: Int = -1
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val v = inflater.inflate(R.layout.fragment_login, container, false)
+        activity?.toolbarTitle?.setText(R.string.greenstand_welcome_text)
+
+        v.sign_up_button?.visibility = View.INVISIBLE
+//Here we will watch the phone edit text to check if the input number(#phoneNumberInput) is in our database already.
+    //if the #phoneNumberInput is in our database we will turn Login button to ActiveButtonStyle and we will set
+    // ClickListener on this button to go on further with the login feature
+     v.phoneEditTextLogin.addTextChangedListener(object: TextWatcher {
+        @SuppressLint("NewApi")
+        override fun afterTextChanged(p0: Editable?) {
+            phoneNumberEntered = p0.toString()
+            if(isExistingUser(phoneNumberEntered!!)) {
+                activateLoginButton()
+            }else {
+                v.emailEditText.addTextChangedListener(object: TextWatcher {
+                    @SuppressLint("NewApi")
+                    override fun afterTextChanged(p0: Editable?) {
+                        emailEntered = p0.toString()
+                        if (isExistingUser(emailEntered!!) || isExistingUser(phoneNumberEntered!!)) {
+                            activateLoginButton()
+                        }else
+                            when (isExistingUser(emailEntered!!)) {
+                                true -> activateLoginButton()
+                                else -> {
+                                    sign_up_button?.visibility = View.VISIBLE
+                                }
+                            }
+                    }
+
+                    override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int){
+                        inactivateLoginButton()
+                    }
+
+                    override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+                        inactivateLoginButton()
+                    }
+                })
+                when (isExistingUser(phoneNumberEntered!!)) {
```

Comment:
This `when` logic is duplicated above us on line 62. To reduce repeating ourselves we can move this logic into a function and pass in any parameters we need such as a phone number or email.